src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/StorageWatcher.cs

```diff
@@ -218,5 +251,47 @@ private async Task StartBlobDownload(bool startup = false)
             
         }
 
+        private async Task StartKustoCacheRefresh(bool startup = false)
+        {
+            var diagConfigRows = await storageService.GetKustoConfiguration();
+            var configsToLoad = new List<DiagConfiguration>();
+            if(!startup)
+            {
+                configsToLoad.AddRange(diagConfigRows.Where(row => row.Timestamp >= kustoCacheLastModifiedTime).ToList());
+            } else
+            {
+                configsToLoad.AddRange(diagConfigRows);
+            }
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
+            try
+            {
+                if (configsToLoad.Count > 0)
+                {
+                    DiagnosticsETWProvider.Instance.LogAzureStorageMessage(nameof(StorageWatcher), $"Starting kusto config download to update cache, Number of configs to load: {configsToLoad.Count}, startup : {startup.ToString()} at {DateTime.UtcNow}");
+                }
+                foreach(var config in configsToLoad)
+                {
+                    var kustoMappingsStringContent = config.KustoClusterMapping;
+                    var kustoMappings = (List<Dictionary<string, string>>)JsonConvert.DeserializeObject(kustoMappingsStringContent, typeof(List<Dictionary<string, string>>));
+                    var resourceProvider = config.RowKey;
+                    if (!kustoCacheService.ContainsKey(resourceProvider) || (kustoCacheService.TryGetValue(resourceProvider, out List<Dictionary<string, string>> value) && !value.Equals(kustoMappings)))
```

Comment:
So for list of dictionaries this `value.Equals(kustoMappings)` doesn't work as we would like unfortunately. I ran a quick test on LINQPad just to confirm.

```
	var map1 = new List<Dictionary<string, string>>{
		new Dictionary<string, string>{
			{"hello", "world"}
		}
	};

	var map2 = new List<Dictionary<string, string>>{
		new Dictionary<string, string>{
			{"hello", "world"}
		}
	};

	Console.Write(map1.Equals(map2));

map1.Equals(map2) == false
```

Reply:
Apparently I tried doing something like this in the KustoMappinsController and I noticed it then too but never had the chance to address it. https://github.com/Azure/Azure-AppServices-Diagnostics/blob/main/src/Diagnostics.RuntimeHost/Controllers/Configuration/KustoClusterMappingsController.cs#L43-L47

Reply:
The kusto map doesn't get updated often so maybe for know just compare if the `Count` is the same. 

```
if (!kustoCacheService.ContainsKey(resourceProvider) || (kustoCacheService.TryGetValue(resourceProvider, out List<Dictionary<string, string>> value) && value.Count == kustoMappings.Count))
```

Reply:
Oh ok. In that case, we can just update it without comparing actual strings. It wont make a difference

Reply:
Here's something you can try `bool isEqual = isEqual = map1.All(m1 => m1.All(kv1 => map2.Any(m2 => m2.Contains(kv1)))) && map1.Count == map2.Count;`